build/Codegen.cmake

```
@@ -93,7 +96,10 @@ function(generate_bindings_for_kernels)
       --tags-path=${site-packages-out}/torchgen/packaged/ATen/native/tags.yaml
       --aten-yaml-path=${site-packages-out}/torchgen/packaged/ATen/native/native_functions.yaml
       --op-selection-yaml-path=${_oplist_yaml}
-  )
+    )
+  if(GEN_ADD_EXCEPTION_BOUNDARY)
+    set(_gen_command "${_gen_command} --add-exception-boundary")
```

I think the set command use semi-colon separator instead of space, i.e. `/opt/conda/envs/py_3.10/bin/python3;-m;torchgen.gen_executorch;--source-path=/pytorch/executorch/configurations/../codegen;--install-dir=/pytorch/executorch/pip-out/temp.linux-x86_64-cpython-310/cmake-out/configurations/optimized_native_cpu_ops_lib;--tags-path=/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torchgen/packaged/ATen/native/tags.yaml;--aten-yaml-path=/opt/conda/envs/py_3.10/lib/python3.10/site-packages/torchgen/packaged/ATen/native/native_functions.yaml;--op-selection-yaml-path=/pytorch/executorch/pip-out/temp.linux-x86_64-cpython-310/cmake-out/configurations/optimized_native_cpu_ops_lib/selected_operators.yaml;--add-exception-boundary`

```suggestion
    set(_gen_command "${_gen_command}" --add-exception-boundary)
```

---

strange, I thought I built this locally. Giving it a shot now.

---

this does seem to have improved the situation (thanks!) but the torchgen version still seems to be wrong: https://github.com/pytorch/executorch/actions/runs/12680876952/job/35343491373?pr=7546#step:9:2186

---

@huydhn still seeing at least some wrong-torchgen-version failures:
- [test-arm-backend-delegation](https://github.com/pytorch/executorch/actions/runs/12699647767/job/35400710061?pr=7546)
- [test-arm-reference-delegation](https://github.com/pytorch/executorch/actions/runs/12699647767/job/35400706527?pr=7546#step:14:1397)
- [test-llava-runner-linux](https://github.com/pytorch/executorch/actions/runs/12699647761/job/35400719004?pr=7546#step:14:8748)
- [test-phi3-mini-runner-linux](https://github.com/pytorch/executorch/actions/runs/12699647761/job/35400719596?pr=7546#step:14:8614)
- [test-eval_llama-wikitext-linux](https://github.com/pytorch/executorch/actions/runs/12699647761/job/35400718418?pr=7546#step:14:8658)
- [test-eval_llama-mmlu-linux](https://github.com/pytorch/executorch/actions/runs/12699647761/job/35400717012?pr=7546#step:14:8644)
- [test-llama_runner_eager-linux](https://github.com/pytorch/executorch/actions/runs/12699647761/job/35400717909?pr=7546#step:14:8625)

however, we do have mac successes. looking into the common thread for these failures now.

---

the ARM one is easy: the workflow directly invokes `install_executorch` from .ci/scripts/utils.sh without "use-pt-pinned-commit". Is there a reason "use-pt-pinned-commit" isn't the default or only option?

test-llava-runner-linux reruns install_requirements.sh and doesn't pass --use-pt-pinned-commit, same with test-phi-3-mini-runner, test-eval_llama-wikitext-linux, test-eval_llama-mmlu-linux, and test-llama_runner_eager-linux. I will send a PR to fix those.